### PR TITLE
[PM-19814] Phishing Detection Warning Popup UI

### PR DESCRIPTION
**Pull request by Banrion**
## 🎟️ Tracking

[PM-19814](https://bitwarden.atlassian.net/browse/PM-19814)

## 📔 Objective

This pull request adds the Phishing Detection feature. This pull request adds updates to the warning popup, a close tab button and a continue button. 

The original code was never merged into main. This branch is created from a previous pull request and branch, updated to be in sync with main, and also adds the UI updates to the warning page.

Previous work: https://github.com/bitwarden/clients/pull/14255

### Changes

- Added close tab button
- Added continue button
- Added dev flag to add "www.test.com" and "www.example.com" to safely test the phishing page
- Made changes to phishing service
    - Track tabs to remove potential edge cases for multiple phishing pages open
    - Added events for content script to interact with service
    - Fixed feature flag enabling
    - Use existing browser api for cross browser compatibility
- Browser api updates
    - Close tab function
    - Navigate tab function

### Follow up changes

- Update non-localized text
- Fix Safari specific bug

### 🎏 Flags

- Enable feature flag on client:  `phishing-detection`
- Enable feature flag on server: `"globalSettings:launchDarkly:flagValues:phishing-detection": "True"`
- Enable dev feature flag for testing custom phishing url:
    - File: `apps/browser/config/local.json` 
```
...
"devFlags": {
    "testPhishingUrls": ["www.example.com"],
 },
...
```
 

### 📝 Notes

- UI jumps when first loading the warning page

## 📸 Screenshots

https://github.com/user-attachments/assets/85368786-11b6-4e22-9775-0b4135e86c7c

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-19814]: https://bitwarden.atlassian.net/browse/PM-19814?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/assets/src/svg/svgs/phishing-landing-icon.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { svgIcon } from "../icon-service";
+
+export const PhishingLandingIcon = svgIcon`
+    <svg width="100" height="100" viewBox="0 0 100 100" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <path d="M56.25 8.33333C56.25 3.73096 59.981 0 64.5833 0H72.9167C77.519 0 81.25 3.73096 81.25 8.33333V12.5H56.25V8.33333Z" fill="#99BAF4"/>
```
Comment (vleague2) [inline libs/assets/src/svg/svgs/phishing-landing-icon.ts]: Yay our first new svg icon in the new lib!! 👏  For these hardcoded colors in this file, can you update them to follow the color guidelines here? https://components.bitwarden.com/?path=/docs/component-library-icon--docs

Comment (willmartian) [reply libs/assets/src/svg/svgs/phishing-landing-icon.ts]: I'm working on an eslint rule to call this out: https://github.com/bitwarden/clients/pull/16167

Comment (Banrion) [reply libs/assets/src/svg/svgs/phishing-landing-icon.ts]: Sure thing! I didn't take a close look at this file and just moved the file after seeing the svg changes from main. I can update it to use the color guidelines!

Comment (vleague2) [reply libs/assets/src/svg/svgs/phishing-landing-icon.ts]: Just an fyi, I realized the icon you are adding here is one that I am in the process of updating in a large icon update ticket (still in progress, no PR yet). If your PR gets merged first that's totally fine, just wanted you to be aware haha

File: apps/browser/src/phishing-detection/background/phishing-detection.service.ts [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+        this._handleNavigationEvent(details);
+      },
+    );
+    // [FIXME] BrowserApi should be used to handle memory leaks in safari but currently
+    // it causes issues with the listener not being called in firefox
+    // BrowserApi.addListener(
+    //   chrome.webNavigation.onCompleted,
+    //   this._handleNavigationEvent.bind(this),
+    // );
+  }
```
Comment (Banrion) [inline apps/browser/src/phishing-detection/background/phishing-detection.service.ts]: ❓ I expected to use the `BrowserApi.addListener` function here. When I set this up and also use the `BrowserApi.sendMessage` function, I do not get the expected results. Any insight would help.

File: apps/browser/src/phishing-detection/background/phishing-detection.service.ts [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+    this._logService = logService;
+    this._storageService = storageService;
+    this._taskSchedulerService = taskSchedulerService;
+
+    logService.info("[PhishingDetectionService] Initialize called");
+
+    configService
+      .getFeatureFlag$(FeatureFlag.PhishingDetection)
+      .pipe(
+        mergeMap(async (enabled) => {
```
Comment (coroiu) [inline apps/browser/src/phishing-detection/background/phishing-detection.service.ts]: suggestion: use `concatMap` to avoid race-conditions when the feature flag updates multiple times in a short time span

File: apps/browser/src/phishing-detection/background/phishing-detection.service.ts (hunk 2 above)
Comment (Banrion) [reply apps/browser/src/phishing-detection/background/phishing-detection.service.ts]: This has been resolved after resetting environment, adding guards for the message type, and binding the function.

File: apps/browser/src/phishing-detection/background/phishing-detection.service.ts (hunk 3 above)
Comment (Banrion) [reply apps/browser/src/phishing-detection/background/phishing-detection.service.ts]: Thanks for noticing this! I've updated it.

File: libs/assets/src/svg/svgs/phishing-landing-icon.ts (hunk 1 above)
Comment (vleague2) [reply libs/assets/src/svg/svgs/phishing-landing-icon.ts]: My [PR for the SVGs ](https://github.com/bitwarden/clients/pull/16219) is in draft but not reviewed yet, so likely yours will make it in first, so we'll want the colors updated to use the tailwind variables!

**Comment by Banrion** [comment]
> Is `apps/browser/src/phishing-detection` the best location for this feature? It looks like we typically put team-scoped folders under `apps/*/src`. Are we intentionally moving away from that?

@willmartian I think this is a good callout. The original code came from an innovation sprint that had not team scoped it yet which I didn't notice. I will make the changes to get this feature in team-scoped folders.

**Review by Patrick-Pimentel-Bitwarden — APPROVED** [review]
👍 Auth changes are fine, probably don't need to be there since it's just an import change from absolute to relative pathing.

File: libs/assets/src/svg/svgs/phishing-landing-icon.ts (hunk 1 above)
Comment (vleague2) [reply libs/assets/src/svg/svgs/phishing-landing-icon.ts]: I merged the icon update PR yesterday, so you should now use the existing `DeactivatedOrg` icon instead of having to add this file!

**Review by Patrick-Pimentel-Bitwarden — APPROVED** [review]

**Review by ttalty — DISMISSED** [review]

**Review by addisonbeck — APPROVED** [review]

File: apps/browser/src/popup/app-routing.module.ts [hunk 4]
```diff
@@ -672,6 +675,32 @@ const routes: Routes = [
...
+          },
+          {
+            path: "",
+            component: LearnMoreComponent,
+            outlet: "secondary",
+          },
+        ],
+        data: {
+          pageIcon: DeactivatedOrg,
+          pageTitle: "Bitwarden blocked it!",
```
Comment (vleague2) [inline apps/browser/src/popup/app-routing.module.ts:697]: I know I said I wasn't going to review because UIF isn't a codeowner anymore 😂 But I just noticed these are not localized? Is that intentional?

Comment (Banrion) [reply apps/browser/src/popup/app-routing.module.ts:697]: No, it isn't intentional. Thanks for calling it out. I'll follow up in a different pull request to fix this. See my following comment on merging for more information on why.

**Comment by Banrion** [comment]
We will be merging this pull request and following up any other improvements and a bug fix for safari with different pull requests. UIF needs this code in main to implement design improvements.

File: apps/browser/src/dirt/phishing-detection/pages/learn-more-component.html [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<span>{{ "phishingPageLearnWhy"| i18n}}</span>
+<a href="http://bitwarden.com/help/phishing-blocked/" bitLink block buttonType="primary">
```
Comment (mKoonrad) [inline apps/browser/src/dirt/phishing-detection/pages/learn-more-component.html:2]: Error 404
